blocks/api/simple-dom.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+
+export function createSimpleNode( node, filter = Array ) {
+	if ( ! node ) {
+		return null;
+	}
+
+	if ( node.nodeType === 3 ) {
+		return node.nodeValue;
+	}
+
+	if ( node.nodeType !== 1 ) {
+		return null;
+	}
+
+	return filter(
+		node.nodeName.toLowerCase(),
+		Array.from( node.attributes || [] ).reduce( ( acc, { name, value } ) => ( {
+			...acc,
+			[ name ]: value,
+		} ), {} ),
+		...createSimpleNodeList( node.childNodes || [], filter )
+	);
+}
+
+export function createSimpleNodeList( nodeList, filter ) {
+	return Array.from( nodeList ).reduce( ( acc, node ) => {
```

Comment:
Micro-optimization / readability: Use [Lodash's `reduce`](https://lodash.com/docs/4.17.4#reduce) directly and skip `Array.from`:

```js
reduce( nodeList, ( acc, node ) => ( {
```